stac_api/runtime/src/links.py

```diff
@@ -0,0 +1,129 @@
+"""A module for injecting links to STAC entries"""
+from typing import Any, Dict
+from urllib.parse import urljoin
+
+import pystac
+
+from fastapi import Request
+from stac_fastapi.types.stac import Collection, Item
+
+from .config import tiles_settings
+from .render import get_render_config
+
+
+class LinkInjector:
+    """
+    A class which organizes information relating STAC entries
+    to endpoints which render associated assets. Used to inject
+    links from catalog entries to tiling endpoints
+
+    ...
+
+    Attributes
+    ----------
+    collection_id : str
+        The ID of a STAC Collection in the PC
+    """
+
+    def __init__(
+        self,
+        collection_id: str,
+        request: Request,
+    ) -> None:
+        """Initialize a LinkInjector"""
+        self.collection_id = collection_id
+        # The collection_id should be suitable for getting a RenderConfig with more details
+        self.render_config = get_render_config()
+        self.tiler_href = tiles_settings.titiler_endpoint or ""
+
+    def inject_collection(self, collection: Collection) -> None:
+        """Inject rendering links to a collection"""
+        collection.get("links", []).append(self._get_collection_map_link())
```

Comment:
🤔

Reply:
Cherry picked this out, so I think this is leftover from the merge. Fixing!